src/network/mod.rs

```diff
@@ -1,46 +1,89 @@
 use async_std::prelude::*;
-use async_std::task::{Context, Poll};
-use core::marker::PhantomData;
-use core::pin::Pin;
+use async_std::stream::{interval, Interval};
+use futures::channel::{mpsc, oneshot};
+use ipfs_embed_db::{BlockStore, Event, Subscription};
 use libipld::block::Block;
-use libipld::codec::Codec;
+use libipld::cid::Cid;
+use libipld::codec::Decode;
 use libipld::error::Result;
-use libipld::multihash::MultihashDigest;
+use libipld::ipld::Ipld;
+use libipld::store::StoreParams;
 use libp2p::core::transport::upgrade::Version;
 use libp2p::core::transport::Transport;
 use libp2p::core::Multiaddr;
 use libp2p::mplex::MplexConfig;
-use libp2p::secio::SecioConfig;
+use libp2p::noise::{Keypair, NoiseConfig, X25519Spec};
 use libp2p::swarm::{Swarm, SwarmEvent};
 use libp2p::tcp::TcpConfig;
 //use libp2p::yamux::Config as YamuxConfig;
-use std::time::Duration;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::time::{Duration, Instant};
 
 mod behaviour;
 mod config;
 
-use crate::storage::{
-    NetworkEvent as StorageEvent, NetworkSubscriber as StorageSubscriber, Storage,
-};
 use behaviour::NetworkBackendBehaviour;
 pub use behaviour::NetworkEvent;
 pub use config::NetworkConfig;
 
-pub struct Network<C: Codec, M: MultihashDigest> {
-    _marker: PhantomData<C>,
-    swarm: Swarm<NetworkBackendBehaviour<M>>,
-    storage: Storage,
-    subscriber: StorageSubscriber,
+pub struct Network<S: StoreParams> {
```

Comment:
bikeshedding; long field names in this struct. Propose
* `bootstrap_completed` -> `booted`
* `subscription` -> `sub`